pkg/controller/pravegacluster/upgrade.go

```diff
@@ -404,6 +405,139 @@ func (r *ReconcilePravegaCluster) syncSegmentStoreVersion(p *pravegav1alpha1.Pra
 	return false, nil
 }
 
+//this function is to check are we doing a rollback in case of a upgrade failure while upgrading from a version below 07 to a version above 07
+func (r *ReconcilePravegaCluster) IsClusterRollbackingFrom07(p *pravegav1alpha1.PravegaCluster) bool {
+	if !r.isRollbackTriggered(p) {
+		return false
+	}
+	//checking if sts above07 exsists
+	stsAbove07 := &appsv1.StatefulSet{}
+	name := util.StatefulSetNameForSegmentstoreAbove07(p.Name)
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: p.Namespace}, stsAbove07)
+	if err != nil && errors.IsNotFound(err) {
+		return false
+	}
+	//checking if sts below07 exsists
+	stsBelow07 := &appsv1.StatefulSet{}
+	name = util.StatefulSetNameForSegmentstoreBelow07(p.Name)
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: p.Namespace}, stsBelow07)
+	if err != nil && errors.IsNotFound(err) {
+		return false
+	}
+	//true will only be return if both sts are present
+	return true
+}
+
+//this function calls the syncstoreversion for upgrade and rollback operation
+func (r *ReconcilePravegaCluster) syncStoreVersion(p *pravegav1alpha1.PravegaCluster) (synced bool, err error) {
+	//in case of a upgrade from a version below 07 to a version above 07 syncSegmentStoreVersionTo07() is called and the util.IsClusterUpgradingTo07(p)  is true in this case
+	//in case of a rollback when upgrade fails from a version below 07 to a version above 07 syncSegmentStoreVersionTo07() is called and the r.IsClusterRollbackingFrom07(p) is true in this case
+	if util.IsClusterUpgradingTo07(p) || r.IsClusterRollbackingFrom07(p) {
+		return r.syncSegmentStoreVersionTo07(p)
+	}
+	//for all other cases of upgrades and rollback this function is called
+	return r.syncSegmentStoreVersion(p)
+}
+
+//this fuction is called when we want to upgrade ss from a version below 07 to a version above 07  as well as when the such upgrades fail then for doing rollback also it's called
+func (r *ReconcilePravegaCluster) syncSegmentStoreVersionTo07(p *pravegav1alpha1.PravegaCluster) (synced bool, err error) {
+	newsts := pravega.MakeSegmentStoreStatefulSet(p)
+	controllerutil.SetControllerReference(p, newsts, r.scheme)
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: newsts.Name, Namespace: p.Namespace}, newsts)
+	//this check is to see if the newsts is present or not if it's not present it will be created here
+	if err != nil && errors.IsNotFound(err) {
```

Comment:
Same as above. Please log all errors. We should not be hiding them.

Reply:
done

Reply:
I don;t see the error being logged????